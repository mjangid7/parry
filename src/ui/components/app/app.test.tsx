import { render, screen } from '@testing-library/react';
import { App } from 'ui/components/app/app';

describe('App component', () => {
  it('renders learn react link', () => {
    render(<App />);
    const linkElement = screen.getByText(/learn react/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('renders logo image', () => {
    render(<App />);
    const logoElement = screen.getByAltText(/logo/i);
    expect(logoElement).toBeInTheDocument();
  });

  it('renders instructional paragraph', () => {
    render(<App />);
    const paragraphElement = screen.getByText(
      /Edit `src\/ui\/components\/app\/app.tsx` and save to reload./i
    );
    expect(paragraphElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
