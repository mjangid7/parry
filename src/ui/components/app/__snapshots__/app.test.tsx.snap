// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app__StyledApp-sc-115d7k0-0 enildR"
  >
    <header
      class="app__Header-sc-115d7k0-1 EyIkl"
    >
      <img
        alt="logo"
        class="app__Logo-sc-115d7k0-2 gxdYOr"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/ui/components/app/app.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="app__Link-sc-115d7k0-3 dvJOTe"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</DocumentFragment>
`;
